include/syslimits.h

```diff
@@ -37,7 +37,7 @@
 #ifndef _SYS_SYSLIMITS_H_
 #define _SYS_SYSLIMITS_H_
 
-#define ARG_MAX 65536 /* max bytes for an exec function */
+#define ARG_MAX 256 /* max bytes for an exec function */
```

Comment:
Value of this constant is very important because it's used by user-space programs as well.

I think you completely missed the fact that this value denotes size of a buffer that stores:
 * `argv` pointer table,
 * argument strings.

**It's not an upper limit for the number of string pointers in `argv` table!**